client/src/lib/attempt-result.js

```diff
@@ -482,6 +483,23 @@ export function attemptResultsWarning(
         };
       }
     }
+
+    // Check whether this result is a duplicate of existing results in the same round.
+    // Excludes FMC and all-DNF results since ties are common.
+    // TODO: Does NOT check the in-progress batch, if any.
+    if (["333fm"].indexOf(eventId) === -1) {
```

Comment:
Above we have `completeAttempts` with filtered-out DNFs, so we can ignore those results upfront:

```suggestion
    if (eventId !== "333fm" && completeAttempts.length > 0) {
```

Then we no longer need the DNF logic in findAllMatchingResults.